src/utils.ts

```diff
@@ -286,7 +286,7 @@ export function castTimestampOrNow(timestamp?: DateTime | string | null): string
     } else if (typeof timestamp === 'string') {
         timestamp = DateTime.fromISO(timestamp)
     }
-    return timestamp.toUTC().toFormat('yyyy-MM-dd HH:mm:ss.u')
+    return timestamp.toUTC().toISO()
```

Comment:
Hmm, isn't the old format required by ClickHouse? At least that's what original ingestion uses

Reply:
Hm... would be great to have E2E clickhouse tests as well to verify this works :). 
I swapped this our since `.toISO()` contains a `Z`, indicating the UTC timezone. The `.toFormat` version contains no timezone information and something at some point re-parses it with my local timezone (`CET`), resulting in some timezone process event test failing.

Reply:
So after this change this will only work with CH if CH has `date_time_input_format` set to `best_effort` instead of the default `basic`. https://clickhouse.tech/docs/en/operations/settings/settings/#settings-date_time_input_format
Originally this function was only used in EE ingestion in Python, but I think w can repurpose it by adding parameter `iso = false` (or something like that) and using it where applicable

Reply:
How about this? https://github.com/PostHog/plugin-server/pull/116/commits/137dcde9e674b1f760543715e360177d6093f187